File: sp67/forum/asm-club/b-cool-w/cpu.c
/* ------------------------------------------------------------------------- */
/* CPU.C  Sample program demonstrating usage of TMIOSDGL(tm) routines        */
/* 									     */
/* Copyright(c) 1994 by B-coolWare.  Written by Bobby Z.		     */
/* ------------------------------------------------------------------------- */
/* files needed to build project:
   CPU.C		- compile with memory model XXX
   CPUTYPE.C		- compile with memory model XXX
   CPU_HL.ASM		- set memory model to XXX,C
   CPUSPEED.ASM		- set memory model to XXX,C
*/

#include <stdio.h>
#include "cputype.h"

void main()
{
 puts("CPU Type Identifier/C  Version 1.14c  Copyright(c) 1994 by B-coolWare.\n");

 printf("  Processor: %s, %dMHz\n",cpuType_Str(),(int)CPU_Speed());
 printf("Coprocessor: %s\n",fpuType_Str());
}

File: sp67/forum/asm-club/b-cool-w/cpu.pas
{$A+,B-,D+,E-,F-,G-,I-,L+,N-,O-,P-,Q-,R-,S-,T-,V-,X+}
{$M 16384,0,655360}
{ --------------------------------------------------------------------------- }
{ CPU.PAS  Sample program demonstrating usage of TMIOSDGL(tm) routines        }
{									      }
{ Copyright(c) 1994 by B-coolWare.  Written by Bobby Z.			      }
{ --------------------------------------------------------------------------- }
{ files needed to build project:

  CPU_HL.ASM
  CPUSPEED.ASM
  CPUTYPE.PAS								      }

uses CPUType;

begin
 WriteLn('CPU Type Identifier/Pas  Version 1.14c  Copyright(c) 1992,94 by B-coolWare.');
 WriteLn;
 WriteLn('  Processor: ',CPU_TypeStr, ', ',Trunc(CPUSpeed),'MHz');
 WriteLn('Coprocessor: ',CoPro_TypeStr);
end.
